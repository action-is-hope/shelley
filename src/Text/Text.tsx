import React, { Ref, forwardRef } from "react";
import type { TextVolume } from "../typings/shared-types";
import { VisuallyHidden } from "../VisuallyHidden/VisuallyHidden";
import { st, classes } from "./text.st.css";

export interface TextProps extends React.HTMLAttributes<HTMLBaseElement> {
  /**
   * HTML as to render as this Text item.
   * @default span
   */
  as: string;
  /** Truncate text to specified line number. */
  truncate?: boolean | number;
  /** Toggles uppercase. */
  uppercase?: boolean;
  /** Set the weight. */
  weight?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | undefined;
  /** Visually hides the element, visible to screen readers. */
  visuallyHidden?: boolean;
  /**
   * How 'loud' should this Text be?
   * @default 3
   */
  vol?: TextVolume;
}

export const Text = forwardRef(
  (
    {
      as: tagName = "span",
      children,
      className: classNameProp,
      truncate = false,
      uppercase = false,
      visuallyHidden,
      weight = undefined,
      vol = 3,
      ...rest
    }: TextProps,
    ref?: Ref<HTMLBaseElement>
  ) => {
    const text = React.createElement(
      tagName,
      {
        className: st(
          classes.root,
          {
            truncate: truncate !== false,
            weight,
            uppercase,
            vol: vol !== false ? vol : undefined,
            lineClamp: typeof truncate === "number" ? truncate : undefined,
          },
          classNameProp
        ),
        ref: ref,
        ...rest,
      },
      children
    );
    return visuallyHidden ? <VisuallyHidden>{text}</VisuallyHidden> : text;
  }
);

Text.displayName = "Text";

type TextExports = Pick<TextProps, Exclude<keyof TextProps, "as">>;

interface TextExportsProps
  extends React.BaseHTMLAttributes<HTMLBaseElement>,
    TextExports {}

export const H1 = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="h1" vol={8} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H1.displayName = "H1";

export const H2 = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="h2" vol={7} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H2.displayName = "H2";

export const H3 = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="h3" vol={6} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H3.displayName = "H3";

export const H4 = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="h4" vol={5} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H4.displayName = "H4";

export const H5 = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="h5" vol={4} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H5.displayName = "H5";

export const H6 = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="h6" vol={4} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H6.displayName = "H6";

export const P = forwardRef(
  ({ children, ...rest }: TextExportsProps, ref?: Ref<HTMLBaseElement>) => (
    <Text as="p" ref={ref} {...rest}>
      {children}
    </Text>
  )
);
P.displayName = "P";
