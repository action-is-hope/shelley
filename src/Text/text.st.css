@namespace "Text";

.root {
  -st-states:
    truncate,
    lineClamp(number),
    weight(number),
    uppercase,
    vol(number);
}

/* Extras */

/* = Uppercase */
.root:uppercase {
  text-transform: uppercase;
}

/* = Truncation */
.root:truncate {
  -webkit-line-clamp: 1;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.root:lineClamp(1) {
  -webkit-line-clamp: 1;
}

.root:lineClamp(2) {
  -webkit-line-clamp: 2;
}

.root:lineClamp(3) {
  -webkit-line-clamp: 3;
}

.root:lineClamp(4) {
  -webkit-line-clamp: 4;
}

.root:lineClamp(5) {
  -webkit-line-clamp: 5;
}

.root:lineClamp(6) {
  -webkit-line-clamp: 6;
}

.root:lineClamp(7) {
  -webkit-line-clamp: 7;
}

.root:lineClamp(8) {
  -webkit-line-clamp: 8;
}

.root:lineClamp(9) {
  -webkit-line-clamp: 9;
}

.root:lineClamp(10) {
  -webkit-line-clamp: 10;
}
