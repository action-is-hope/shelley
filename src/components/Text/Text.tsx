/* Text.tsx */
import React from "react";
import { TextVolume } from "../types";
import style from "./text.st.css";
import classNames from "classnames";
import VisuallyHidden from "../VisuallyHidden/VisuallyHidden";

export interface TextProps extends React.HTMLAttributes<HTMLBaseElement> {
  /** HTML tag to render as this Text item. */
  tag: string;
  /** Single line truncation - requires a 'known' width. */
  truncate?: boolean;
  /** Toggles uppercase styles. */
  uppercase?: boolean;
  /** Set the weight index. */
  weight?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | false;
  /** Visually hides the element, visible to screen readers. */
  visuallyHidden?: boolean;
  /** How 'loud' should this Text be? */
  vol?: TextVolume;
}

const Text = React.forwardRef(
  (
    {
      tag: tagName,
      children,
      className,
      truncate = false,
      uppercase,
      visuallyHidden,
      weight = false,
      vol = 3,
      ...rest
    }: TextProps,
    ref?: React.Ref<HTMLBaseElement>
  ) => {
    const rootClassNames = classNames(
      style.root,
      style["vol" + vol],
      className,
      {
        uppercase
      }
    );

    const text = React.createElement(
      tagName,
      {
        ...style(rootClassNames, { truncate, weight }, rest),
        ref: ref,
        ...rest
      },
      children
    );
    return visuallyHidden ? <VisuallyHidden>{text}</VisuallyHidden> : text;
  }
);

Text.displayName = "Text";

type TextExports = Pick<TextProps, Exclude<keyof TextProps, "tag">>;

interface TestProps
  extends React.BaseHTMLAttributes<HTMLBaseElement>,
    TextExports {}

export const H1 = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="h1" vol={8} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H1.displayName = "H1";

export const H2 = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="h2" vol={7} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H2.displayName = "H2";

export const H3 = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="h3" vol={6} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H3.displayName = "H3";

export const H4 = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="h4" vol={5} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H4.displayName = "H4";

export const H5 = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="h5" vol={4} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H5.displayName = "H5";

export const H6 = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="h6" vol={4} ref={ref} {...rest}>
      {children}
    </Text>
  )
);
H6.displayName = "H6";

export const P = React.forwardRef(
  ({ children, ...rest }: TestProps, ref?: React.Ref<HTMLBaseElement>) => (
    <Text tag="p" ref={ref} {...rest}>
      {children}
    </Text>
  )
);
P.displayName = "P";

export default Text;
