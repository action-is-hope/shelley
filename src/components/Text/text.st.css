/** text.st.css */
@namespace "Text";

.root {
  -st-states:
    truncate,
    weight(enum(1, 2, 3, 4, 5, 6, 7, 8)),
    uppercase,
    vol(enum(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12));
}

/* Extras */

.root:truncate {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}
