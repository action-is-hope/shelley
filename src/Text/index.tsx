import { Text, TextProps } from "./Text";
import { forwardRef, Ref } from "react";

export type TextExports = Omit<TextProps, "as">;

export * from "./Text";

function H1(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="h1" ref={ref} vol={8} {...rest}>
      {children}
    </Text>
  );
}
/**
 * H1 is a page title and there should only be one on the page.
 */
const _H1 = forwardRef(H1);
H1.displayName = "H1";
export { _H1 as H1 };

function H2(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="h2" ref={ref} vol={7} {...rest}>
      {children}
    </Text>
  );
}
/**
 * H2 is a page heading level 2.
 */
const _H2 = forwardRef(H2);
H2.displayName = "H2";
export { _H2 as H2 };

function H3(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="h3" ref={ref} vol={6} {...rest}>
      {children}
    </Text>
  );
}
/**
 * H3 is a page heading level 3.
 */
const _H3 = forwardRef(H3);
H3.displayName = "H3";
export { _H3 as H3 };

function H4(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="h4" ref={ref} vol={5} {...rest}>
      {children}
    </Text>
  );
}
/**
 * H4 is a page heading level 4.
 */
const _H4 = forwardRef(H4);
H4.displayName = "H4";
export { _H4 as H4 };

function H5(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="h5" ref={ref} vol={4} {...rest}>
      {children}
    </Text>
  );
}
/**
 * H5 is a page heading level 5.
 */
const _H5 = forwardRef(H5);
H5.displayName = "H5";
export { _H5 as H5 };

function H6(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="h6" ref={ref} vol={4} {...rest}>
      {children}
    </Text>
  );
}
/**
 * H6 is a page heading level 6.
 */
const _H6 = forwardRef(H6);
H6.displayName = "H6";
export { _H6 as H6 };

function P(props: TextExports, ref: Ref<HTMLBaseElement>) {
  const { children, ...rest } = props;
  return (
    <Text as="p" ref={ref} {...rest}>
      {children}
    </Text>
  );
}
/**
 * P is a paragrapgh.
 */
const _P = forwardRef(P);
P.displayName = "P";
export { _P as P };
