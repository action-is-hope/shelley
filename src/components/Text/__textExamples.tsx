/* textExamples.tsx */
import React from "react";
// Example Component.
import Text from "./Text";
import Grid from "../../components/Grid/Grid";

// Supporting Components.
import PropsDemo from "../../components_site/PropsDemo/PropsDemo";
import CodeSample from "../../components_site/CodeSample/CodeSample";
import { P } from "../../components/Text/Text";

// Styles.
import grid from "../../projects/default/css/grid.st.css";

export const meta = {
  name: "Button"
};

export const QuickRef = () => (
  <CodeSample className={grid.mid}>
    {`import { Text, P, H1, H2, H3, H4, H5, H6 } from "@action-is-hope/shelley";

<H1 vol={8}>Shortcuts for html headings, h1-h6, set your own volume.</H1>
  
<P>The mighty paragraph, defaults to a volume of 3.</P>
<Text tag="span">Text with your choice of tag via the "tag" prop.</Text>`}
  </CodeSample>
);

// Props Demo - Examples below.
export const ComponentDemo = () => {
  const [TextDemoProps, setTextDemoProps]: any = React.useState([
    {
      name: "vol",
      label: "vol",
      type: "number",
      min: 1,
      max: 12,
      value: 3
    },
    {
      name: "truncate",
      label: "truncate",
      type: "checkbox",
      value: false
    },
    {
      name: "uppercase",
      label: "uppercase",
      type: "checkbox",
      value: false
    },
    {
      name: "children",
      label: "children",
      type: "textarea",
      value:
        // "The IPCC prepares comprehensive Assessment Reports about knowledge on climate change, its causes, potential impacts and response options. The IPCC also produces Special Reports, which are an assessment on a specific issue and Methodology Reports, which provide practical guidelines for the preparation of greenhouse gas inventories."
        "IPCC, how can I explain it, I'll take you frame by frame it. Shelley likes to rap about climate change reports in the hope that people might read them, go figure. Shelley also thanks Naughty By Nature for their contribution to music and head bopping."
    }
  ]);

  return (
    <PropsDemo
      id="textPropsDemo"
      demoProps={TextDemoProps}
      setDemoProps={setTextDemoProps}
      tsExtends="HTMLBaseElement"
      // className={grid.goal}
      renderExample={
        <CodeSample>{`import { P } from "@action-is-hope/shelley";\n<P ${
          TextDemoProps[0].value ? `\n  vol={${TextDemoProps[0].value}}` : ``
        } ${TextDemoProps[1].value ? `\n  truncate` : ``} ${
          TextDemoProps[3].value !== "none" ? `\n  uppercase` : ``
        } \n>\n  ${TextDemoProps[3].value}\n</P>`}</CodeSample>
      }
    >
      <Text
        id="textDemo"
        tag="p"
        vol={TextDemoProps[0].value}
        truncate={TextDemoProps[1].value}
        uppercase={TextDemoProps[2].value}
      >
        {TextDemoProps[3].value}
      </Text>
      <br />
      <br />
      <Text tag="p" vol={3}>
        Just in case:{" "}
        <a href="https://www.ipcc.ch/reports/">
          <abbr title="Intergovernmental Panel on Climate Change">IPCC</abbr>{" "}
          reports
        </a>{" "}
        üåç
      </Text>
    </PropsDemo>
  );
};

export const ComponentHTML = () => <CodeSample>{`TBC`}</CodeSample>;

export const ComponentCSS = () => (
  <CodeSample className={grid.mid} fixedHeight>{`/** text.st.css */

  :import {
    -st-from: "../../../components/Text/text.st.css";
    -st-default: Text;
  }
  
  /* Volume */
  Text.vol1 {}
  Text.vol2 {}
  Text.vol3 {}
  Text.vol4 {}
  Text.vol5 {}
  Text.vol6 {}
  Text.vol7 {}
  Text.vol8 {}
  Text.vol9 {}
  Text.vol10 {}
  Text.vol11 {}
  Text.vol12 {}
  
  /* Extras */
  Text.uppercase {}
  Text.underline {}
  Text.bold {}
  
  /* State selectors - Note: Inherits functional styles from Shelley. */
  Text:truncate {}
  
  /* Whatever you need which you can use as classNames by importing your text.st.css */
  Text.hint {}
  Text.error {}
  Text.taxonomy-color1 {}
  Text.TupacOrBiggie {}
  `}</CodeSample>
);

export const Example1 = () => (
  // We have just used P for the demo, use vol as the design requires and the tags for semantics.
  <Grid variant={1}>
    <P vol={1} data-test="your-id">
      vol 1. Text
    </P>
    <P vol={2}>vol 2. Text</P>
    <P>vol 3. Text</P>
    <P vol={4}>vol 4. Text</P>
    <P vol={5}>vol 5. Text</P>
    <P vol={6}>vol 6. Text</P>
    <P vol={7}>vol 7. Text</P>
    <P vol={8}>vol 8. Text</P>
    <P vol={9}>vol 9. Text</P>
    <P vol={10}>
      vol 10. Text <br />
      <br />
    </P>
  </Grid>
);
