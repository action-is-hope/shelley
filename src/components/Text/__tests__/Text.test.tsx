import Text, { P, H1, H2, H3, H4, H5, H6 } from "../Text";
import renderer from "react-test-renderer";

const longText = "The cryosphere refers to frozen components of the Earth system. Around 10% of Earth's land area is covered by glaciers or ice sheets. The ocean and cryosphere support unique habitats, and are interconnected with other components of the climate system through global exchange of water, energy and carbon."
const shortText = "I am Groot!"

describe("H", () => {
  test("H1", () => {
    const tree = renderer
      .create(
        <H1>{shortText}</H1>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H2", () => {
    const tree = renderer
      .create(
        <H2>{shortText}</H2>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H3", () => {
    const tree = renderer
      .create(
        <H3>{shortText}</H3>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H3 volume", () => {
    const tree = renderer
      .create(
        <H3 vol={5}>{shortText}</H3>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H4", () => {
    const tree = renderer
      .create(
        <H4>{shortText}</H4>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H5", () => {
    const tree = renderer
      .create(
        <H5>{shortText}</H5>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H6", () => {
    const tree = renderer
      .create(
        <H6>{shortText}</H6>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("H uppercase", () => {
    const tree = renderer
      .create(
        <H1 uppercase>{shortText}</H1>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("P", () => {
  test("P", () => {
    const tree = renderer
      .create(
        <P>{shortText}</P>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("P volume", () => {
    const tree = renderer
      .create(
        <P vol={5}>{shortText}</P>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("P vol weight", () => {
    const tree = renderer
      .create(
        <P vol={8} weight={2}>
          vol 8 Text
        </P>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
  test("P truncate", () => {
    const tree = renderer
      .create(
        <P truncate={2}>
          {longText}
        </P>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Text", () => {
  test("Text", () => {
    const tree = renderer
      .create(
        <Text as="h1">{shortText}</Text>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Text volume", () => {
    const tree = renderer
      .create(
        <Text as="h3" vol={6}>{shortText}</Text>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Text as truncate vol", () => {
    const tree = renderer
      .create(
        <Text as="div" truncate={6} vol={false}>
          <P>
            {longText}
          </P>

          <P>
            The cryosphere refers to frozen components of the Earth system. Around 10% of Earth&apos;s land area is covered by glaciers or ice sheets. The ocean and cryosphere support unique habitats, and are interconnected with other components of the climate system through global exchange of water, energy and carbon.
          </P>
        </Text>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

