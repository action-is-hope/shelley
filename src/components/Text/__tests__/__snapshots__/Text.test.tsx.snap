// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H H uppercase 1`] = `
<h1
  className="Text__Test__root Text__Test--uppercase Text__Test---vol-1-8"
>
  I am Groot!
</h1>
`;

exports[`H H1 1`] = `
<h1
  className="Text__Test__root Text__Test---vol-1-8"
>
  I am Groot!
</h1>
`;

exports[`H H2 1`] = `
<h2
  className="Text__Test__root Text__Test---vol-1-7"
>
  I am Groot!
</h2>
`;

exports[`H H3 1`] = `
<h3
  className="Text__Test__root Text__Test---vol-1-6"
>
  I am Groot!
</h3>
`;

exports[`H H3 volume 1`] = `
<h3
  className="Text__Test__root Text__Test---vol-1-5"
>
  I am Groot!
</h3>
`;

exports[`H H4 1`] = `
<h4
  className="Text__Test__root Text__Test---vol-1-5"
>
  I am Groot!
</h4>
`;

exports[`H H5 1`] = `
<h5
  className="Text__Test__root Text__Test---vol-1-4"
>
  I am Groot!
</h5>
`;

exports[`H H6 1`] = `
<h6
  className="Text__Test__root Text__Test---vol-1-4"
>
  I am Groot!
</h6>
`;

exports[`P P 1`] = `
<p
  className="Text__Test__root Text__Test---vol-1-3"
>
  I am Groot!
</p>
`;

exports[`P P truncate 1`] = `
<p
  className="Text__Test__root Text__Test--truncate Text__Test---vol-1-3 Text__Test---lineClamp-1-2"
>
  The cryosphere refers to frozen components of the Earth system. Around 10% of Earth's land area is covered by glaciers or ice sheets. The ocean and cryosphere support unique habitats, and are interconnected with other components of the climate system through global exchange of water, energy and carbon.
</p>
`;

exports[`P P vol weight 1`] = `
<p
  className="Text__Test__root Text__Test---weight-1-2 Text__Test---vol-1-8"
>
  vol 8 Text
</p>
`;

exports[`P P volume 1`] = `
<p
  className="Text__Test__root Text__Test---vol-1-5"
>
  I am Groot!
</p>
`;

exports[`Text Text 1`] = `
<h1
  className="Text__Test__root Text__Test---vol-1-3"
>
  I am Groot!
</h1>
`;

exports[`Text Text as truncate vol 1`] = `
<div
  className="Text__Test__root Text__Test--truncate Text__Test---lineClamp-1-6"
>
  <p
    className="Text__Test__root Text__Test---vol-1-3"
  >
    The cryosphere refers to frozen components of the Earth system. Around 10% of Earth's land area is covered by glaciers or ice sheets. The ocean and cryosphere support unique habitats, and are interconnected with other components of the climate system through global exchange of water, energy and carbon.
  </p>
  <p
    className="Text__Test__root Text__Test---vol-1-3"
  >
    The cryosphere refers to frozen components of the Earth system. Around 10% of Earth's land area is covered by glaciers or ice sheets. The ocean and cryosphere support unique habitats, and are interconnected with other components of the climate system through global exchange of water, energy and carbon.
  </p>
</div>
`;

exports[`Text Text volume 1`] = `
<h3
  className="Text__Test__root Text__Test---vol-1-6"
>
  I am Groot!
</h3>
`;
